### PR TITLE
add support for linkvertise pastes

### PR DESCRIPTION
**Pull request by davidkra230**
Fixes (Links to issues fixed by this PR): 
None

Description:
bypasses linkvertise's pasting feature. basically i just added a check for pastes and if it matches it will get the info almost exactly like the url bypass except it copies the paste instead of redirecting.

Test links:
[LinkvertisePaste](https://linkvertise.com/114822/test?o=sharing)
Checklist:

- [x] I made sure there are no unnecessary changes in the code*
- [x] Tested on Chromium- Browser OS
- [ ] Tested on Firefox

### CODE REVIEW

File: src/js/injection_script.js [hunk 1]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+	    const rawOpen = XMLHttpRequest.prototype.open
+
+	    XMLHttpRequest.prototype.open = function () {
+		this.addEventListener("load", (data) => {
+		    if (data.currentTarget.responseText.includes("tokens")) {
+			const response = JSON.parse(data.currentTarget.responseText)
+			if (!response.data.valid)
+			    return insertInfoBox(
+				"Please solve the captcha, afterwards we can immediately redirect you"
+			    )
```
Comment (lem6ns) [inline src/js/injection_script.js:2847]: why was formatting changed here?

File: src/js/injection_script.js [hunk 2]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+
+			const target_token = response.data.tokens["TARGET"]
+			const ut = localStorage.getItem("X-LINKVERTISE-UT")
+			const linkvertise_link = location.pathname.replace(
+			    /\/[0-9]$/,
+			    ""
+			)
```
Comment (lem6ns) [inline src/js/injection_script.js:2854]: why was formatting changed here?

File: src/js/injection_script.js [hunk 3]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+
+			fetch(
+			    `https://publisher.linkvertise.com/api/v1/redirect/link/static${linkvertise_link}?X-Linkvertise-UT=${ut}`
+			)
+			    .then((r) => r.json())
+			    .then((json) => {
+				if (json?.data.link.target_type !== "URL") {
+				    insertInfoBox("Paste detected, bypassing..")
```
Comment (lem6ns) [inline src/js/injection_script.js:2862]: this line isn't needed, and what if `.target_type` was also not a paste?

File: src/js/injection_script.js [hunk 4]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+				    // what is referred to as the pathmap is a table of functions to bypass for each type of content.
+				    let pathmap = {
```
Comment (lem6ns) [inline src/js/injection_script.js:2864]: change to `const` instead of `let`

File: src/js/injection_script.js [hunk 5]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+					"PASTE": function () {
+					    if (json?.data.link.id) {
+						const json_body = {
+						    serial: btoa(
+							JSON.stringify({
+							    timestamp:
+								new Date().getTime(),
+							    random: "6548307",
+							    link_id: json.data.link.id,
+							})
```
Comment (lem6ns) [inline src/js/injection_script.js:2874]: consider making an inner function for creating a serial

File: src/js/injection_script.js [hunk 6]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+						}
+						fetch(
+						    `https://publisher.linkvertise.com/api/v1/redirect/link${linkvertise_link}/paste?X-Linkvertise-UT=${ut}`,
+						    {
+							method: "POST",
+							body: JSON.stringify(json_body),
+							headers: {
+							    Accept: "application/json",
+							    "Content-Type":
+								"application/json",
```
Comment (lem6ns) [inline src/js/injection_script.js:2886]: awkward formatting?

File: src/js/injection_script.js [hunk 7]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+						    }
+						)
+						    .then((r) => r.json())
+						    .then((json) => {
+							if (json?.data.paste) {
+							    insertInfoBox("Bypassed, attempting to copy to clipboard.")
+							    navigator.clipboard.writeText(json?.data.paste).then(function() {
+							      insertInfoBox("Copied paste to clipboard!\nYou may now close this tab.")
+							    }, (err) => {
+							      window.prompt("Copying to clipboard failed!\nThe paste is displayed below. Ctrl+C to copy it.", json?.data.paste)
```
Comment (lem6ns) [inline src/js/injection_script.js:2897]: consider creating a blob text file, then creating an objectURL to redirect/open a new tab to.

File: src/js/injection_script.js [hunk 8]
```diff
@@ -2828,59 +2828,126 @@ ensureDomLoaded(() => {
...
+				    return pathmap[json?.data.link.target_type]()
+				}
+				if (json?.data.link.id) {
+				    const json_body = {
+					serial: btoa(
+					    JSON.stringify({
+						timestamp: new Date().getTime(),
+						random: "6548307",
+						link_id: json.data.link.id,
+					    })
```
Comment (lem6ns) [inline src/js/injection_script.js:2922]: consider making an inner function for creating a serial

**Review by lem6ns — CHANGES_REQUESTED** [review]
please fix formatting and see review comments.

File: src/js/injection_script.js (hunk 1 above)
Comment (davidmeower) [reply src/js/injection_script.js:2847]: prettier was used

File: src/js/injection_script.js (hunk 2 above)
Comment (davidmeower) [reply src/js/injection_script.js:2854]: prettier was used

File: src/js/injection_script.js (hunk 6 above)
Comment (davidmeower) [reply src/js/injection_script.js:2886]: prettier was used

**Comment by davidkra230** [comment]
I have fixed these but, I had deleted the original repository in which I had made the changes and therefore, I need to open a new pull request.

I will close this pull request for now, and I will open a new one with the requested changes.

**Comment by lem6ns** [comment]
alright

**Pull request closed without merging**